src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,57 +3,164 @@
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
- * Represents a Person's address in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
+ * Represents a Person's address in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    private static final int LENGTH_OF_CORRECT_ADDRESS = 4;
+    public static final String EXAMPLE = "123, some street, #510-341, 493494";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person address must be in the format : block, street, unit, postalcode\n"
+            + "where block street and unit are nonempty strings while postal\n"
+            + "code is a 6 digit number. They must be separated by commas";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_POSTALCODE_VALIDATION_REGEX = "\\d{6}";
+    private static final String ADDRESS_UNIT_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_STREET_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_BLOCK_VALIDATION_REGEX = ".+";
+
+    private AddressBlock addressBlock;
+    private AddressStreet addressStreet;
+    private AddressUnit addressUnit;
+    private AddressPostalcode addressPostalcode;
 
-    public final String value;
     private boolean isPrivate;
 
     /**
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException
+     *             if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        initialisePersonData(trimmedAddress);
+    }
+
+
+    /**
+     * Initialises the attributes of the Address object by splitting the string using comma.
+     */
+    private void initialisePersonData(String trimmedAddress) throws IllegalValueException {
+        String[] personDataSplit = trimmedAddress.split(",");
+        if (personDataSplit.length != 4) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        this.addressBlock = new AddressBlock(personDataSplit[0]);
+        this.addressStreet = new AddressStreet(personDataSplit[1]);
+        this.addressUnit = new AddressUnit(personDataSplit[2]);
+        this.addressPostalcode = new AddressPostalcode(personDataSplit[3]);
```

Comment:
Can more refactoring be done to make the code more readable? 
Say refactoring out magic numbers?